packages/function-entry/src/input.jsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import debug from 'debug';
+import classNames from 'classnames';
+import range from 'lodash/range';
+import MuiInput from 'material-ui/Input';
+import TextField from 'material-ui/TextField';
+import { MuiThemeProvider, createMuiTheme, withStyles } from 'material-ui/styles';
+import { FormControl, FormHelperText } from 'material-ui/Form';
+import { indicators } from '@pie-lib/render-ui';
+import PropTypes from 'prop-types';
+
+const { Correct, Incorrect, NothingSubmitted } = indicators;
+
+const tags = {
+  'correct': Correct,
+  'incorrect': Incorrect,
+  'empty': NothingSubmitted
+}
+
+const log = debug('pie-elements:function-entry:input');
+
+class RawInput extends React.Component {
+  static propTypes = {
+    dark: PropTypes.bool,
+    classes: PropTypes.object,
+    correctness: PropTypes.oneOf(['correct', 'incorrect', 'empty']),
+    disabled: PropTypes.bool,
+    error: PropTypes.string,
+    inputProps: PropTypes.object,
+    onChange: PropTypes.func,
+    size: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    value: PropTypes.string.isRequired,
+    feedback: PropTypes.string
+  }
+  render() {
+    const {
+      dark,
+      classes,
+      correctness,
+      disabled,
+      error,
+      inputProps,
+      onChange,
+      size,
+      value,
+      feedback
+    } = this.props;
+
+    const formClasses = classNames(classes.formControl);
+    const inputClass = classNames(classes.input, classes[`size${size}`]);
+    const CorrectnessTag = tags[correctness];
+    const theme = createMuiTheme({
+      palette: {
+        type: dark ? 'dark' : 'light'
+      },
+    });
+
+
+    return (
+      <FormControl
+        disabled={disabled}
+        className={formClasses}
+        error={!!error} >
+        <div className={classes.inputAndIcon}>
+          <MuiThemeProvider theme={theme}>
+            <MuiInput
+              classes={{
+                root: classes.inputRoot,
+                input: inputClass
+              }}
+              value={value}
+              onChange={onChange}
+              inputComponent={TextField}
+              inputProps={inputProps}  />
+          </MuiThemeProvider>
+          {CorrectnessTag && <div className={classes.icon}><CorrectnessTag feedback={feedback || 'feedback'} /></div>}
```

Comment:
falling back to 'feedback' here? if feedback is undefined|null then it'll just render the icon, so should be good with that.